go/array_string/two_sum/solution.go

```diff
@@ -0,0 +1,23 @@
+// Package solution contains the implementation of
+// the two sum algorithm solution with O(n) complexity
+// and also an amortized O(1) complexity.
+package solution
```

Comment:
Less important for this, but when you write actual packages think about their names (and their exported-value names) from the perspective of a user. For example, a package called `util` doesn't tell you anything at all about what it contains. https://golang.org/doc/effective_go.html#package-names is a good reference